core_backend/app/question_answer/routers.py

```diff
@@ -104,18 +104,23 @@ async def search(
         exclude_archived=exclude_archived,
     )
 
-    if type(response) is QueryResponse:
-        await save_query_response_to_db(user_query_db, response, asession)
```

Comment:
We were running the same lines for both types so moved it out. This does mean that it will error if the type is not `QueryResponse` or `QueryResponseError` even before we get to the ValueError line

Reply:
Though in theory that should never happen

Reply:
yeah fair enough... let's move the check up then (like if not either of the two types return ValueError, else move on)? Or drop it altogether?

Reply:
Ok will do